### PR TITLE
Refactor(eos_designs): Wildcard dict to list for connected_endpoints_keys

### PR DESCRIPTION
**Pull request by ghost**
## Data model key

`connected_endpoints_keys`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1: Claus
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2:
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/defaults/main/default-connected-endpoints-keys.yml [hunk 1]
```diff
@@ -1,17 +1,17 @@
 default_connected_endpoints_keys:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/defaults/main/default-connected-endpoints-keys.yml]: I don't think we are ready to modify the defaults yet. I suggest you copy the updated data-model to the `all.yml` for the `eos_designs_unit_tests_v4.0` scenario, and restore the original in this file.

Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/defaults/main/default-connected-endpoints-keys.yml]: okay

Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/defaults/main/default-connected-endpoints-keys.yml]: fixed

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by emilarista — APPROVED** [review]
LGTM!

**Pull request merged**